### ISSUE TITLE
loss is nan

### ISSUE THREAD

**Issue by happy20200** [open-mmlab/mmocr#836]
![image](https://user-images.githubusercontent.com/59427046/158114264-ed321d6c-6ff6-4da2-8922-e89daac30a58.png)
训练没几轮loss就开始是nan，精度就开始是1，这是怎么回事啊（训练的数据是自己的数据，只有一个类别）

**Comment by happy20200** [comment]
_base_ = [
    '../../_base_/runtime_10e.py',
    '../../_base_/det_models/ocr_mask_rcnn_r50_fpn_ohem.py',
    '../../_base_/schedules/schedule_sgd_160e.py',
    '../../_base_/det_datasets/icdar2015.py',
    '../../_base_/det_pipelines/maskrcnn_pipeline.py'
]

train_list = {{_base_.train_list}}
test_list = {{_base_.test_list}}

train_pipeline = {{_base_.train_pipeline}}
test_pipeline_icdar2015 = {{_base_.test_pipeline_icdar2015}}

data = dict(
    samples_per_gpu=8,
    workers_per_gpu=4,
    val_dataloader=dict(samples_per_gpu=1),
    test_dataloader=dict(samples_per_gpu=1),
    train=dict(
        type='UniformConcatDataset',
        datasets=train_list,
        pipeline=train_pipeline),
    val=dict(
        type='UniformConcatDataset',
        datasets=test_list,
        pipeline=test_pipeline_icdar2015),
    test=dict(
        type='UniformConcatDataset',
        datasets=test_list,
        pipeline=test_pipeline_icdar2015))

evaluation = dict(interval=10, metric='hmean-iou')

#####################
# optimizer
# optimizer = dict(type='SGD', lr=0.08, momentum=0.9, weight_decay=0.0001)
optimizer = dict(type='SGD', lr=0.001, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
# learning policy
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=0.0001,
    # warmup_ratio=0.001,
    step=[80, 128])
# total_epochs = 160
total_epochs = 100
我将学习率调整之后，loss暂时没有出现nan的情况，但是测试的时候什么都检测不到，请问问题可能出现在哪里啊

**Comment by gaotongxiao** [comment]
The learning rate is way lower than the suggested one, meaning that the model also converges much slower than expected. You can compare your actual loss with [ours] (https://download.openmmlab.com/mmocr/textdet/maskrcnn/mask_rcnn_r50_fpn_160e_icdar2015_20210219-8eb340a3.log.json) to estimate your progress. You may try increasing the learning rate a bit for faster convergence.

**Comment by happy20200** [comment]
![image](https://user-images.githubusercontent.com/59427046/158140125-df4cbff9-fef0-4ee9-b143-cb53ab16a7ab.png)
这是十轮的结果，这，，

**Comment by happy20200** [comment]
> The learning rate is way lower than the suggested one, meaning that the model also converges much slower than expected. You can compare your actual loss with [ours] (https://download.openmmlab.com/mmocr/textdet/maskrcnn/mask_rcnn_r50_fpn_160e_icdar2015_20210219-8eb340a3.log.json) to estimate your progress. You may try increasing the learning rate a bit for faster convergence.

但是我学习率调大之后，loss就会变成nan

**Comment by xinke-wang** [comment]
Have you visualized the annotations of the custom data to ensure the converted labels are correct?

**Comment by happy20200** [comment]
还没有，请问哪里有现成的工具可以进行可视化吗

**Comment by xinke-wang** [comment]
If you use [mmocr.utils.convert_annotations](https://github.com/open-mmlab/mmocr/blob/main/mmocr/utils/data_convert_util.py) to convert your dataset, it would be easy to visualize the labels by using some tools designed for MS COCO format annotations.

Or you can just insert some visualization functions into your data converters and draw the bounding boxes to see if their behaviors are as you expected.

**Comment by happy20200** [comment]
发现问题了，就是标签没有做对，我之前转换的方式不对，打算用[mmocr.utils.convert_annotations]来做，不过里面的image_infos应该是什么样子的啊
![image](https://user-images.githubusercontent.com/59427046/158153024-9a5a6973-b376-4043-9ba2-e9f381c400d8.png)

**Comment by xinke-wang** [comment]
You can check this [funsd_convert.py](https://github.com/open-mmlab/mmocr/blob/main/tools/data/textdet/funsd_converter.py) as an example.

https://github.com/open-mmlab/mmocr/blob/main/tools/data/textdet/funsd_converter.py#L78-L82

**Comment by gaotongxiao** [comment]
I'm closing this issue, but feel free to reopen it if you have any followup questions.

**Comment by xReniar** [comment]
@gaotongxiao I want to reopen this issue because I'm having the same problem. I'm trying to train `crnn`, I have controlled all my training data and there is nothing wrong. Could the problem be due to the resize of the images?

This are my configuration files:
sroie_textrecog_data_root = "data/sroie"
```
sroie_textrecog_train = dict(
    type="OCRDataset",
    data_root = sroie_textrecog_data_root,
    ann_file = "textrecog_train.json",
    pipeline = None
)

sroie_textrecog_test = dict(
    type="OCRDataset",
    data_root = sroie_textrecog_data_root,
    ann_file = "textrecog_val.json",
    test_mode = True,
    pipeline = None
)
```
```
_base_ = [
    "../_base_/datasets/sroie.py",
    "../_base_/default_runtime.py",
    "../_base_/schedules/schedule_adadelta_5e.py",
    "_base_crnn_mini-vgg.py",
]

train_list = [_base_.sroie_textrecog_train]
test_list = [_base_.sroie_textrecog_test]
default_hooks = dict(logger=dict(type='LoggerHook', interval=50))

train_dataloader = dict(
    batch_size=1,
    num_workers=1,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type='ConcatDataset',
        datasets=train_list,
        pipeline=_base_.train_pipeline))

val_dataloader = dict(
    batch_size=1,
    num_workers=1,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type='ConcatDataset',
        datasets=test_list,
        pipeline=_base_.test_pipeline))
test_dataloader = val_dataloader

val_evaluator = dict(dataset_prefixes=['Sroie'])
test_evaluator = val_evaluator
```
